test/models/deploy_spec_test.rb

```diff
@@ -684,10 +684,104 @@ class DeploySpecTest < ActiveSupport::TestCase
       assert_equal 'node_modules/.bin/lerna publish --yes --skip-git --repo-version 1.0.0 --force-publish=* --npm-tag latest', @spec.deploy_steps[1]
     end
 
+    test '#valid_publish_config? is false when publishConfig is missing in package_json and ENFORCE_PUBLISH_CONFIG is set' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"name": "foo"}')
+      ENV['ENFORCE_PUBLISH_CONFIG'] = '1'
+
+      @spec.expects(:package_json).returns(package_json)
+      refute @spec.valid_publish_config?
+    end
+
+    test '#valid_publish_config? is true when ENFORCE_PUBLISH_CONFIG is unset' do
+      ENV['ENFORCE_PUBLISH_CONFIG'] = nil
+      assert @spec.valid_publish_config?
+    end
+
+    test '#publish_config returns publishConfig in package.json' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": "foo"}')
+
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal "foo", @spec.publish_config
+    end
+
+    test '#valid_publish_config_access? is false when publishConfig.access is invalid' do
+      @spec.stubs(:publish_config_access).returns('foo')
+      refute @spec.valid_publish_config_access?
+    end
+
+    test '#valid_publish_config_access? is true when publishConfig.access is public or restricted' do
+      @spec.stubs(:publish_config_access).returns('public')
+      assert @spec.valid_publish_config_access?
+
+      @spec.stubs(:publish_config_access).returns('restricted')
+      assert @spec.valid_publish_config_access?
+    end
+
+    test '#publish_config_access is restricted when publishConfig.access is missing' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": {}}')
+
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal 'restricted', @spec.publish_config_access
+    end
+
+    test '#publish_config_access returns publishConfig.access in package.json' do
+      package_json = Pathname.new('/tmp/fake_package.json')
+      package_json.write('{"publishConfig": {
+          "access": "foo"
+        }
+      }')
+      @spec.expects(:package_json).returns(package_json)
+      assert_equal "foo", @spec.publish_config_access
+    end
+
+    test "#package_scoped_when_private? is false when private packages are not scoped" do
+      @spec.stubs(:scoped_package?).returns(false)
+      @spec.stubs(:publish_config_access).returns("restricted")
+      refute @spec.package_scoped_when_private?
+    end
+
+    test "#package_scoped_when_private? is true when private packages are scoped" do
+      @spec.stubs(:scoped_package?).returns(true)
+      @spec.stubs(:publish_config_access).returns("restricted")
+      assert @spec.package_scoped_when_private?
+    end
+
+    test "#package_scoped_when_private? is true when public packages are not scoped" do
```

Comment:
This highlights the incomprehensibility of the method name.  The name talks about private packages, but it also provides results for public packages?